acceptance/builder_test.go

```diff
@@ -0,0 +1,415 @@
+				"--env", "CNB_BUILDPACKS_DIR=/env_folders/different_buildpack_dir_from_env",
+				builderImage,
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNil(t, err)
+			expected := "CNB_BUILDPACK_DIR: /env_folders/different_buildpack_dir_from_env"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("CNB_LAYERS_DIR is set", func() {
+		it("CNB_LAYERS_DIR is successfully transmitted to build script", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+				"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+				"--env", "CNB_LAYERS_DIR=/layers/different_layer_dir_from_env",
+				builderImage,
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNil(t, err)
+			expected := "LAYERS_DIR: /layers/different_layer_dir_from_env/hello_world"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("CNB_PLAN_PATH is set", func() {
+		it("provides the buildpack a filtered version of the plan found at CNB_PLAN_PATH", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+				"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/different_plan_from_env.toml",
+				builderImage,
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNil(t, err)
+			expected := "different_plan_from_env.toml_reqires_subset_content"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("CNB_PLATFORM_DIR is set", func() {
+		it("CNB_PLATFORM_DIR is successfully transmitted to build script", func() {
+			command := exec.Command(
+				"docker",
+				"run",
+				"--rm",
+				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+				"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+				"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+				"--env", "CNB_PLATFORM_DIR=/env_folders/different_platform_dir_from_env",
+				builderImage,
+			)
+			output, err := command.CombinedOutput()
+			h.AssertNil(t, err)
+			expected := "PLATFORM_DIR: /env_folders/different_platform_dir_from_env"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+}
+
+func assertBuilderMetadata(t *testing.T, path string) {
+	contents, _ := ioutil.ReadFile(path)
+	h.AssertEq(t, len(contents) > 0, true)
+
+	var analyzedMd platform.BuildMetadata
+	_, err := toml.Decode(string(contents), &analyzedMd)
+	h.AssertNil(t, err)
+}
+
+func getBuilderMetadata(t *testing.T, path string) *platform.BuildMetadata {
+	contents, _ := ioutil.ReadFile(path)
+	h.AssertEq(t, len(contents) > 0, true)
+
+	var analyzedMd platform.BuildMetadata
+	_, err := toml.Decode(string(contents), &analyzedMd)
+	h.AssertNil(t, err)
+
+	return &analyzedMd
+}
```

Comment:
These two functions are almost the same.
I think it's better to not have duplicated code.
Moreover, I'm not sure why you would want to only `assertBuilderMetadata` without checking the returned metadata from `getBuilderMetadata`.
So probably it's better to keep only `getBuilderMetadata`.

Reply:
fixed in 284958efb31b1a468aa4d0daa153680cb8287c67